rabbitmq_queue_master_balancer/rabbitmq_queue_master_balancer.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+
+import logging
+import time
+
+from configargparse import ArgumentParser
+from prettylog import basic_config, LogFormat
+from pyrabbit2.api import Client
+from pyrabbit2.http import NetworkError
+
+parser = ArgumentParser(auto_env_var_prefix='RQMB_')
+
+parser.add_argument(
+    '--api-url',
+    type=str,
+    default='localhost:15672'
+)
+
+parser.add_argument(
+    '--user',
+    type=str,
+    default='guest'
+)
+
+parser.add_argument(
+    '--password',
+    type=str,
+    default='guest'
+)
+
+parser.add_argument(
+    '-L', '--log-level', help='Log level',
+    default='info',
+    choices=(
+        'critical', 'fatal', 'error', 'warning',
+        'warn', 'info', 'debug', 'notset'
+    ),
+)
+
+parser.add_argument(
+    '--log-format', choices=LogFormat.choices(), default=LogFormat.stream
+)
+
+parser.add_argument(
+    '--dry-run',
+    action="store_true",
+    help='Show how many queues will move from one node (with a maximum queues) \
+    to another (with a minimum queues)'
+)
+
+parser.add_argument(
+    '--queue-delta',
+    type=int,
+    default=3,
+    help='Reasonable delta of queues between max and min numbers of queues \
+    on node when script do nothing'
+)
+
+parser.add_argument(
+    '--relocate-queues',
+    type=int,
+    default=2,
+    help='How many queues will relocate in one time'
+)
+
+arguments = parser.parse_args()
+
+log = logging.getLogger()
+
+
+def wait_for_client(client):
+    '''
+    :param client:
+    :return: bool
+    '''
+
+    log.debug('RabbitMQ http not ready, waiting...')
+    try:
+        return client.is_alive()
+    except NetworkError:
+        time.sleep(5)
+        wait_for_client(client)
+
+
+def nodes_dict(nodes_info_data, vhost_names: list):
+    '''
+    :param client
+    :param vhost_names list of vhosts
+    :return: dict: Key is a name of rabbit node, Value is empty list
+    '''
+    nodes_dictionary = dict.fromkeys(
```

Comment:
very complex part for reading, probably using temporary dictionary and returning it will be better.

Reply:
+